src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComImportInteropInterfaceDetailsStrategy.cs

```diff
@@ -0,0 +1,192 @@
+    /// [assembly:IgnoresAccessChecksTo("AssemblyContainingArgType2")]
+    /// // One attribute per containing assembly of each type used in each method signature of the interface.
+    ///
+    /// namespace System.Runtime.CompilerServices
+    /// {
+    ///     [AssemblyUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    ///     internal class IgnoresAccessChecksToAttribute : Attribute
+    ///     {
+    ///         public IgnoresAccessChecksToAttribute(string assemblyName) { }
+    ///     }
+    /// }
+    ///
+    /// [DynamicInterfaceCastableImplementation]
+    /// interface InterfaceForwarder : IComInterface
+    /// {
+    ///     RetType IComInterface.Method1(ArgType1 arg1, ArgType2 arg2, ...)
+    ///     {
+    ///         return ((IComInterface)((IComImportAdapter)this).GetRuntimeCallableWrapper())(arg1, arg2, ...);
+    ///     }
+    /// }
+    /// </code>
+    ///
+    /// This mechanism allows source-generated COM interop to allow using runtime-implemented COM interfaces with runtime-defined marshalling behavior with minimal work on the source-generated COM interop side.
+    /// Additionally, by scoping the majority of the logic to this class, we make this logic more easily trimmable.
+    ///
+    /// We emit the <c>IgnoresAccessChecksToAttribute</c> to enable casting to internal <see cref="ComImportAttribute"/> types, which is a very common scenario (most <see cref="ComImportAttribute"/> types are internal).
+    /// </remarks>
+    [RequiresDynamicCode("Enabling interop between source-generated and built-in COM is not supported when trimming is enabled.")]
+    [RequiresUnreferencedCode("Enabling interop between source-generated and built-in COM requires dynamic code generation.")]
+    internal sealed class ComImportInteropInterfaceDetailsStrategy : IIUnknownInterfaceDetailsStrategy
+    {
+        public static readonly IIUnknownInterfaceDetailsStrategy Instance = new ComImportInteropInterfaceDetailsStrategy();
+
+        private readonly ConditionalWeakTable<Type, Type> _forwarderInterfaceCache = new();
+
+        // TODO: Support exposing ComImport interfaces through StrategyBasedComWrappers?
+        public IComExposedDetails? GetComExposedTypeDetails(RuntimeTypeHandle type) => DefaultIUnknownInterfaceDetailsStrategy.Instance.GetComExposedTypeDetails(type);
+
+        public IIUnknownDerivedDetails? GetIUnknownDerivedDetails(RuntimeTypeHandle type)
+        {
+            Type runtimeType = Type.GetTypeFromHandle(type)!;
+            if (!runtimeType.IsImport)
+            {
+                return DefaultIUnknownInterfaceDetailsStrategy.Instance.GetIUnknownDerivedDetails(type);
+            }
+
+            Type implementationType = _forwarderInterfaceCache.GetValue(runtimeType, runtimeType =>
+            {
+                AssemblyBuilder assembly = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("ComImportForwarder"), AssemblyBuilderAccess.RunAndCollect);
```

Comment:
```suggestion
                AssemblyBuilder assembly = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("ComImportForwarder"), runtimeType.IsCollectible ? AssemblyBuilderAccess.RunAndCollect : AssemblyBuilderAccess.Run);
```
This should reduce the overhead quite a bit for typical non-collectible use case.